chains/evm/contracts/test/onRamp/OnRamp/OnRamp.addressEncodingCompatibility.t.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IPoolV1} from "../../../interfaces/IPool.sol";
+import {IPoolV2} from "../../../interfaces/IPoolV2.sol";
+
+import {Client} from "../../../libraries/Client.sol";
+import {ExtraArgsCodec} from "../../../libraries/ExtraArgsCodec.sol";
+import {MessageV1Codec} from "../../../libraries/MessageV1Codec.sol";
+import {Pool} from "../../../libraries/Pool.sol";
+import {OnRamp} from "../../../onRamp/OnRamp.sol";
+import {OnRampSetup} from "./OnRampSetup.t.sol";
+
+import {IERC165} from "@openzeppelin/contracts@5.3.0/utils/introspection/IERC165.sol";
+import {VmSafe} from "forge-std/Vm.sol";
+
+contract OnRamp_addressEncodingCompatibility is OnRampSetup {
+  bytes32 internal constant CCIP_MESSAGE_SENT_TOPIC =
+    keccak256("CCIPMessageSent(uint64,uint64,bytes32,address,bytes,(address,uint32,uint32,uint256,bytes)[],bytes[])");
+
+  function decode(
+    bytes calldata encodedMessage
+  ) external pure returns (MessageV1Codec.MessageV1 memory) {
+    return MessageV1Codec._decodeMessageV1(encodedMessage);
+  }
+
+  function _forwardAndDecode(
+    Client.EVM2AnyMessage memory message,
+    address originalSender
+  ) internal returns (MessageV1Codec.MessageV1 memory) {
+    uint256 fee = s_onRamp.getFee(DEST_CHAIN_SELECTOR, message);
+
+    deal(s_sourceFeeToken, address(s_onRamp), type(uint96).max);
+    vm.startPrank(address(s_sourceRouter));
+    vm.recordLogs();
+    s_onRamp.forwardFromRouter(DEST_CHAIN_SELECTOR, message, fee, originalSender);
+    vm.stopPrank();
+
+    VmSafe.Log[] memory logs = vm.getRecordedLogs();
+    bytes memory encodedMessage;
+    for (uint256 i = 0; i < logs.length; ++i) {
+      if (
+        logs[i].emitter == address(s_onRamp) && logs[i].topics.length != 0
```

Comment:
nit: I don't think we need to check the onramp as emitter as there's nothing that would be able to emit this log in the tests

Reply:
Fixed [4364168](https://github.com/smartcontractkit/chainlink-ccip/pull/1476/commits/4364168607e32569355c39a87c88e9903557e7f4)